### PR TITLE
Add query_regions to esa.hsa

### PR DESCRIPTION
**Pull request by keflavich**
The ESA HSA tool should have a `query_regions` tool to match it to the general astroquery API.  This is my suggestion for how to do that.

It also passes on some kwargs that could be useful.

### CODE REVIEW

File: astroquery/esa/hsa/core.py [hunk 1]
```diff
@@ -405,7 +407,40 @@ def query_observations(self, coordinate, radius, *, n_obs=10):
                  f"where contains("
                  f"point('ICRS', hsa.v_active_observation.ra, hsa.v_active_observation.dec), "
                  f"circle('ICRS', {coord.ra.degree},{coord.dec.degree},{r.to(u.deg).value}))=1")
-        return self.query_hsa_tap(query)
+        return self.query_hsa_tap(query, **kwargs)
```
Comment (lvalerom) [inline astroquery/esa/hsa/core.py:410]: Maybe here is cleaner to just call `query_region`:

```python
return self.query_region(coordinate, radius, n_obs=n_obs, columns="observation_id", **kwargs)
```

File: astroquery/esa/hsa/core.py [hunk 2]
```diff
@@ -405,7 +407,40 @@ def query_observations(self, coordinate, radius, *, n_obs=10):
...
+
```
Comment (lvalerom) [inline astroquery/esa/hsa/core.py:411]:
```suggestion
```

**Review by lvalerom — CHANGES_REQUESTED** [review]
Looks good to me. I would add a test case for the new function in `hsa_test.py`:

```python
def test_query_region(self):
    c = SkyCoord(ra=100.2417*u.degree, dec=9.895*u.degree, frame='icrs')
    parameters = {'coordinate': c,
                  'radius': 0.5}
    dummyTapHandler = DummyHSATapHandler("query_region", parameters)
    hsa = HSAClass(self.get_dummy_tap_handler())
    hsa.query_region(**parameters)
    dummyTapHandler.check_call("query_region", parameters)
```

**Comment by keflavich** [comment]
@lvalerom you can submit your suggested change as a PR onto this one.  That's the best way to make sure you get credit for those code changes.  Your suggestions are all good.

**Review by lvalerom — APPROVED** [review]

**Review by bsipocz — APPROVED** [review]

**Comment by bsipocz** [comment]
OK, so including the narrative docs we have a decent 84% coverage including the query_region/observations methods, so I go ahead and merge this now.

Thank you both!

**Pull request merged**